medusa/reconstruct/expand.py

```diff
@@ -285,137 +97,68 @@ def _continuous_iterative_binary_gapfill(model,phenotype_dict,cycle_order,
     variables = chain(*reaction_variables)
     for variable in variables:
         coefficients[variable] = 0.0
+    # update the linear coefficients for the added reaction to the penalitties of the reaction
+    new_coefficients =  {}
+    for reaction in [rxn.id for rxn in gapfiller.reactions]:
+        if reaction in reaction_probability.keys():
+            new_coefficients[reaction] = 1-reaction_probability[reaction]
+    for react_id in new_coefficients.keys():
+        reaction = gapfiller.reactions.get_by_id(react_id)
+        for_var = reaction.forward_variable
+        rev_var = reaction.reverse_variable
+        if coefficients[for_var]>0.0:
+            coefficients[for_var] = new_coefficients[react_id]
+        if coefficients[rev_var]>0.0:
+            coefficients[rev_var] = new_coefficients[react_id]
     gapfiller.objective.set_linear_coefficients(coefficients)
-
-    ## set a constraint on flux through the original objective
-    for reaction in original_objective.keys():
-        print("Constraining lower bound for " + reaction)
-        gapfiller.reactions.get_by_id(reaction).lower_bound = lower_bound
-
-    exchange_reactions = [rxn for rxn in gapfiller.reactions if\
-                            rxn.id.startswith(exchange_prefix)]
-    for rxn in exchange_reactions:
-        rxn.lower_bound = 0
-
-    for cycle_num in range(0,output_ensemble_size):
-        print("starting cycle number " + str(cycle_num))
+    
+    # Generate the coefficients list and Randomly varying the coefficients 
+    iou = []
+    for y in coefficients.values():
+        iou.append(y)
+    solutions = []
+    t = 0.05
+    for n in range(100):
+        new_coeff = iou + np.random.normal(0, t, len(iou))
+        coefficients = dict(zip(coefficients.keys(), abs(new_coeff)))
+                            
+        gapfiller.objective.set_linear_coefficients(coefficients)
+        for reaction in original_objective.keys():
+            print("Constraining lower bound for " + reaction)
+            gapfiller.reactions.get_by_id(reaction).lower_bound = lower_bound
         cycle_reactions = set()
-        original_coefficients = \
-            gapfiller.objective.get_linear_coefficients(gapfiller.variables)
-
-        for condition in cycle_order[cycle_num]:
-            # set the medium for this condition.
-            for ex_rxn in phenotype_dict[condition].keys():
-                gapfiller.reactions.get_by_id(ex_rxn).lower_bound = \
-                    -1.0*phenotype_dict[condition][ex_rxn]
-                gapfiller.reactions.get_by_id(ex_rxn).upper_bound = \
-                    1.0*phenotype_dict[condition][ex_rxn]
```

Comment:
We still need this for the original function to work--make sure you add this back in when you transfer your new code over to gapfill_to_ensemble